### ISSUE TITLE
[IAM drift] Resources in `DELETE_REQUESTED` state should be filtered out

### ISSUE THREAD

**Issue by gjonathanhong** [abcxyz/guardian#334]
### TL;DR

Context for https://github.com/abcxyz/guardian/blob/main/pkg/commands/drift/drift.go#L194. 

Currently, `asset.SearchAllIamPolicies` will include IAM policies bound to resources that are soft-deleted (state=`DELETE_REQUESTED`) and these are identified as drift. These are false-positives and should be filtered out.

**Initial approach**
1. Get all soft-deleted projects/folders with `asset.SearchAllResources`
2. Exclude them using a query for `asset.SearchAllIamPolicies`. 

However, I found that `asset.SearchAllIamPolicies` does not support negated queries, which makes it infeasible to exclude projects/folders in `DELETE_REQUESTED` state from the API response. 

**A hacky alternative**
1. `SearchAllResources` for folders/projects in `state:DELETE_REQUESTED`
2. Append these resources to the driftignore hierarchy graph so they will be filtered out using the driftignore approach

While the alternative could work, the best solution is if we can apply the filter in the `SearchAllIamPolicies` request, which doesn't seem to be possible at the moment.

### Environment Details

All
